Source/WTF/wtf/CompactVariantOperations.h

```diff
@@ -59,17 +59,24 @@ template<typename T> concept  CompactVariantAlternative =
     || CompactVariantAlternativeSmallEnough<T>
     || CompactVariantTraits<T>::hasAlternativeRepresentation;
 
+// A CompactVariant stores data by bit-packing the variant index
+// and data into a storage of type Storage (uint64_t):
+// * Bit 63-56: if pointer/smart pointer, bit 63-56 of the pointer. Useful
+//              if the architecture supports top-byte ignore, and information
+//              is stored there. Otherwise zeroes.
+// * Bit 55-48: variant index (indicates the data type of the variant)
+// * Bit 47-0 : if pointer/smart pointer, bit 48-0 of the pointer. If other
+//              types of data, the encoded data.
+// This struct provides operations on the packed storage in a CompactVariant.
 template<CompactVariantAlternative... Ts> struct CompactVariantOperations {
     using StdVariant = Variant<Ts...>;
     using Index = uint8_t;
     using Storage = uint64_t;
     static constexpr Storage movedFromDataValue = std::numeric_limits<Storage>::max();
-    static constexpr Storage totalSize = sizeof(Storage) * 8;
-    static constexpr Storage indexSize = sizeof(Index) * 8;
-    static constexpr Storage indexShift = totalSize - indexSize;
-    static constexpr Storage payloadSize = totalSize - indexSize;
-    static constexpr Storage payloadMask = (1ULL << payloadSize) - 1;
-    static_assert(payloadSize + indexSize <= totalSize);
+    static constexpr Storage indexShift = 48;
+    static constexpr Storage indexMask = 0xffULL << indexShift;
```

Comment:
The `ULL` here isn't needed because the shift count is of type `Storage` and that causes promotion of the other operand.

Reply:
The compiler doesn't seem to like that, so I'll keep the `ULL`

```
CompactVariantOperations.h:77:30: error: constexpr variable 'indexMask' must be initialized by a constant expression
   77 |     static constexpr Storage indexMask = 0xFF << indexShift;
CompactVariantOperations.h:77:47: note: shift count 48 >= width of type 'int' (32 bits)
   77 |     static constexpr Storage indexMask = 0xFF << indexShift;
```